eth-client/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
+name = "eth-client"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+tokio = { version ="1.35.1",  features = ["full"] }
+ethcontract = "0.25.4"
+serde = "1.0.195"
+serde_json = "1.0.111"
+primitive-types = "0.12.2"
+web3 = { version = "0.19", default-features = false, features = ["signing"] }
```

Comment:
Please put all the deps except `ethcontract` and `web3` inside root `Cargo.toml`